server/src/test/java/org/elasticsearch/indices/state/RareClusterStateIT.java

```diff
@@ -397,6 +399,21 @@ public void onFailure(Exception e) {
 
         assertBusy(() -> assertTrue(client().prepareGet("index", "type", "1").get().isExists()));
 
+        ActionFuture<IndexResponse> dynamicMappingsFut = client().prepareIndex("index", "type", "2").setSource("field2", 42).execute();
```

Comment:
I presume this just biffs up the test, or am I missing something specific?

Reply:
I also wonder if we sometime should not have an extra field

Reply:
the first document that is put on the test does not use dynamic mappings (there is an explicit putMapping call for that one). This one does an actual dynamic mapping update (the thing we changed in this PR).
Not having an extra field is just what the first document tests :)

Reply:
I've added a comment to the test here to clarify

Reply:
Ok, thanks. I thought that the first call was indeed dynamic (and also adds a type, which is why another call to only add a field is required). Thanks for clarifying.